proxy/grpcproxy/cache/store.go

```diff
@@ -73,7 +77,11 @@ func (c *cache) Add(req *pb.RangeRequest, resp *pb.RangeResponse) {
 	defer c.mu.Unlock()
 
 	if req.Revision > c.compactedRev {
-		c.lru.Add(key, resp)
+		if req.Revision == 0 {
+			c.lru.Set(key, resp, defaultCurrentTTL)
+		} else {
+			c.lru.Set(key, resp, defaultHistoricTTL)
+		}
```

Comment:
Historic keys are immutable so the TTL can be indefinite. Keys on the current revision are subject to change, so expiring on the TTL will refresh against the current server state.